core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestionBuilder.java

```diff
@@ -478,12 +498,18 @@ public static SuggestMode resolve(final String str) {
     /** An enum representing the valid sorting options */
     public enum SortBy implements Writeable<SortBy> {
         /** Sort should first be based on score, then document frequency and then the term itself. */
-        SCORE,
+        SCORE((byte) 0x0),
         /** Sort should first be based on document frequency, then score and then the term itself. */
-        FREQUENCY;
+        FREQUENCY((byte) 0x1);
 
         protected static SortBy PROTOTYPE = SCORE;
 
+        private byte id;
+
+        SortBy(byte id) {
```

Comment:
Maybe we don't need this byte representation if we write the ordinals. TermSuggestion could read and write the same way we send SortBy over the wire, but only if this isn't breaking bwc. Again (sorry ;-)), maybe @areek can comment as well?